app/uk/gov/hmrc/agentaccesscontrol/service/MtdAuthorisationService.scala

```diff
@@ -24,16 +26,37 @@ import uk.gov.hmrc.play.http.HeaderCarrier
 import scala.concurrent.{ExecutionContext, Future}
 
 class MtdAuthorisationService(agenciesConnector: AgenciesConnector,
-                              relationshipsConnector: RelationshipsConnector) {
+                              relationshipsConnector: RelationshipsConnector,
+                              auditService: AuditService) extends LoggingAuthorisationResults {
 
-  def authoriseForSa(agentCode: AgentCode, mtdSaClientId: MtdSaClientId)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Boolean] = {
+  def authoriseForSa(agentCode: AgentCode, mtdSaClientId: MtdSaClientId)
+                    (implicit ec: ExecutionContext, hc: HeaderCarrier, request: Request[_]): Future[Boolean] = {
     agenciesConnector.fetchAgencyRecord(agentCode) flatMap  {
-      case Some(agency) => hasRelationship(agency.arn, mtdSaClientId)
-      case None => Future successful false
+      case Some(agency) => hasRelationship(agency.arn, mtdSaClientId) map { result =>
+        auditDecision(agentCode, mtdSaClientId, result, "arn" -> agency.arn)
```

Comment:
at the moment this line can be changed to `auditDecision(agentCode, mtdSaClientId, true, "arn" -> agency.arn)`, i.e. hard coding result=true, without a test failing, adding unit tests checks similar to AuthorisationService's would probably make a test fail with result hard coded to true.